pkg/config/factory.go

```diff
@@ -446,16 +457,34 @@ func (k *kubeConfigFactory) ParseConfig(ctx context.Context,
 			Name:      meta.Name,
 			Namespace: meta.Namespace,
 		}
+
+		contextOption := pkgcuex.WithExtraData("context", contextValue)
+		parameterOption := pkgcuex.WithExtraData("parameter", meta.Properties)
+		// Compile the config template
+		val, err := cuex.KubeVelaDefaultCompiler.CompileStringWithOptions(context.TODO(), string(template.Template), contextOption, parameterOption)
+		if err != nil {
+			return nil, fmt.Errorf("failed to compile config template: %w", err)
+		}
+
+		// Render the validation response
+		valid, err := val.LookupPath(cuelang.ParsePath(SaveTemplateKey + ".validation")).MarshalJSON()
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse config template validation: %w", err)
+		}
+		validation := Validation{}
+		if err = json.Unmarshal(valid, &validation); err != nil {
+			return nil, fmt.Errorf("failed to parse template validation into validation: %w", err)
+		}
+
 		// Render the output secret
-		output, err := template.Template.RunAndOutput(contextValue, meta.Properties)
-		if err != nil && !cue.IsFieldNotExist(err) {
-			return nil, err
+		output, err := val.LookupPath(cuelang.ParsePath(SaveTemplateKey + "output")).MarshalJSON()
```

Comment:
Similar above. We do not need to MarshalJSON first and then decode it. cue.Value supports decoding directly (which calls MarshalJSON inside).